static/js/recent_topics.js

```diff
@@ -0,0 +1,72 @@
+var recent_topics = (function () {
+
+var exports = {};
+
+var topics = [];
```

Comment:
You should consider using our `Dict` class here, which can handle case-insensitive keys and has a simple set/get/has API.